### PR TITLE
Add com.outerwildsmods.owmods_gui

### PR DESCRIPTION
**Pull request by Bwc9876**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. Link: https://github.com/Bwc9876/ow-mod-man
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.outerwildsmods.owmods_gui.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+app-id: com.outerwildsmods.owmods_gui
+runtime: org.gnome.Platform
+runtime-version: '42'
```
Comment (hfiguiere) [inline com.outerwildsmods.owmods_gui.yml:3]:
```suggestion
runtime-version: '44'
```

42 is EOL

File: com.outerwildsmods.owmods_gui.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        - tar xf data.tar.gz --directory deb_data/
+        - install -Dm755 deb_data/usr/bin/outer-wilds-mod-manager /app/bin/outer-wilds-mod-manager
+        - install -Dm755 deb_data/usr/share/applications/outer-wilds-mod-manager.desktop /app/share/applications/com.outerwildsmods.owmods_gui.desktop
+        - install -Dm644 -t /app/share/metainfo/ com.outerwildsmods.owmods_gui.metainfo.xml
+        - mkdir -p /app/share/icons/hicolor
+        - cp -r deb_data/usr/share/icons/hicolor/* /app/share/icons/hicolor/
+        - rm -rf /app/share/icons/hicolor/512x512@2x
+    sources:
+      - type: file
+        url: https://github.com/Bwc9876/ow-mod-man/releases/download/gui_v0.5.0/outer-wilds-mod-manager_0.5.0_amd64.deb
```
Comment (hfiguiere) [inline com.outerwildsmods.owmods_gui.yml:38]: you still build an aarch64 flatpak

File: com.outerwildsmods.owmods_gui.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+runtime: org.gnome.Platform
+runtime-version: '42'
+sdk: org.gnome.Sdk
+sdk-extensions:
+    - org.freedesktop.Sdk.Extension.mono6
+command: outer-wilds-mod-manager
+rename-icon: outer-wilds-mod-manager
+finish-args:
+    - --socket=fallback-x11
+    - --socket=wayland
```
Comment (hfiguiere) [inline com.outerwildsmods.owmods_gui.yml:11]: does it really support wayland?

File: com.outerwildsmods.owmods_gui.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    - --share=ipc
+    - --share=network
+    - --filesystem=home
+    - --filesystem=/tmp:rw
+    - --filesystem=~/.steam/steam/steamapps/common/Outer Wilds/:rw
```
Comment (hfiguiere) [inline com.outerwildsmods.owmods_gui.yml:16]: you don't need this if you already request `home` which is broad.

**Comment by hfiguiere** [comment]
bot, build com.outerwildsmods.owmods_gui

File: com.outerwildsmods.owmods_gui.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2023 Bwc9876 -->
+<component type="desktop-application">
+    <id>com.outerwildsmods.owmods_gui</id>
```
[the remark this answers is not in the document]
Comment (Bwc9876) [reply com.outerwildsmods.owmods_gui.metainfo.xml:4]: Yes, we're currently in the process of creating a new mod manager. The [owmods GitHub org](https://github.com/ow-mods) owns the domain, and I'm a part of that org.

File: com.outerwildsmods.owmods_gui.yml (hunk 3 above)
Comment (Bwc9876) [reply com.outerwildsmods.owmods_gui.yml:11]: I'm basing this manifest off of [another app that uses tauri](https://github.com/flathub/in.cinny.Cinny/blob/master/in.cinny.Cinny.yml), which has that marked. And I've tested it on Wayland itself under plasma.

File: com.outerwildsmods.owmods_gui.yml (hunk 1 above)
Comment (Bwc9876) [reply com.outerwildsmods.owmods_gui.yml:3]: The application framework I'm using, Tauri, requires an older version of libwebkit2gtk that the 44 version doesn't provide

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "skip-arches": ["aarch64"]
```
Comment (hfiguiere) [inline flathub.json:2]:
```suggestion
  "only-arches": ["x86_64"]
```

File: com.outerwildsmods.owmods_gui.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    - org.freedesktop.Sdk.Extension.mono6
+command: outer-wilds-mod-manager
+rename-icon: outer-wilds-mod-manager
+finish-args:
+    - --socket=fallback-x11
+    - --socket=wayland
+    - --share=ipc
+    - --share=network
+    - --filesystem=home
+    - --filesystem=/tmp:rw
```
Comment (hfiguiere) [inline com.outerwildsmods.owmods_gui.yml:15]:
```suggestion
    - --filesystem=/tmp
```

default is rw

File: com.outerwildsmods.owmods_gui.yml (hunk 1 above)
Comment (hfiguiere) [reply com.outerwildsmods.owmods_gui.yml:3]: the problem really is 42 is EOL. So maybe fixing the tauri platform?

**Comment by hfiguiere** [comment]
bot, build com.outerwildsmods.owmods_gui

File: com.outerwildsmods.owmods_gui.yml (hunk 1 above)
Comment (barthalion) [reply com.outerwildsmods.owmods_gui.yml:3]: We don't allow new apps using EOL'ed runtimes. Have you tried bundling older libwebkit2gtk?

Comment (hfiguiere) [reply com.outerwildsmods.owmods_gui.yml:3]: that include an older libsoup. (it's in the shared modules).

Comment (Bwc9876) [reply com.outerwildsmods.owmods_gui.yml:3]: I'll work on getting webkit2gtk 40 in the bundle, thanks for the tip about libsoup!

Comment (Bwc9876) [reply com.outerwildsmods.owmods_gui.yml]: Upgraded the runtime to GNOME 44

**Comment by barthalion** [comment]
bot, build com.outerwildsmods.owmods_gui

**Comment by hfiguiere** [comment]
bot, build com.outerwildsmods.owmods_gui

File: com.outerwildsmods.owmods_gui.yml (hunk 3 above)
Comment (Bwc9876) [reply com.outerwildsmods.owmods_gui.yml:11]: I decided to switch back to Wayland to test this and yes it does work without issue.

File: com.outerwildsmods.owmods_gui.yml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+          url: https://webkitgtk.org/releases/
+          version-pattern: LATEST-STABLE-(\d[\.\d]+\d)
+          url-template: https://webkitgtk.org/releases/webkitgtk-$version.tar.xz
+      - type: shell
+        commands:
+          - sed -i 's@NAMES avif.h@NAMES avif/avif.h@' Source/cmake/FindAVIF.cmake
+          - sed -i '/PATH_SUFFIXES avif/d' Source/cmake/FindAVIF.cmake
+          - sed -i 's@${AVIF_INCLUDE_DIR}/avif.h@${AVIF_INCLUDE_DIR}/avif/avif.h@'
+            Source/cmake/FindAVIF.cmake
+  - name: openssl11
```
Comment (hfiguiere) [inline com.outerwildsmods.owmods_gui.yml:46]: do you really need this old version of openssl?

Comment (Bwc9876) [reply com.outerwildsmods.owmods_gui.yml:46]: Yes it's a requirement imposed by Tauri's HTTP client

File: com.outerwildsmods.owmods_gui.yml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+sdk-extensions:
+    - org.freedesktop.Sdk.Extension.mono6
+command: outer-wilds-mod-manager
+rename-icon: outer-wilds-mod-manager
+finish-args:
+    - --socket=fallback-x11
+    - --socket=wayland
+    - --share=ipc
+    - --share=network
+    - --filesystem=home
```
Comment (barthalion) [inline com.outerwildsmods.owmods_gui.yml:14]: Why does it need full home access, instead of just the path to Outer Wilds?

Comment (Bwc9876) [reply com.outerwildsmods.owmods_gui.yml:14]: In order to keep backwards compatibility with our old manager, the app needs to access `~/.local/share/OuterWildsModManager`. However I see that `home` is too broad and updated the manifest accordingly.

**Comment by hfiguiere** [comment]
bot, build com.outerwildsmods.owmods_gui

**Pull request closed without merging**